### PR TITLE
Vendor forge-fmt into solang repo

### PR DESCRIPTION
**Pull request by seanyoung**
The published forge-fmt is out date, not maintained and depends on an old version of solang-parser.

This is a copy of forge-fmt:

 https://github.com/foundry-rs/foundry/tree/master/crates/fmt

commit: 00854b602ef0e67379a2027ccc5d0aad553e5333

### CODE REVIEW

**Comment by seanyoung** [comment]
> I don't know if this is much of an effort, but have you considered adding forge-fmt as a submodule or as a git subtree?

It needs modifications to make it work, and I don't know that that is possible with submodule or subtree

File: fmt/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "forge-fmt"
+version = "0.2.0"
```
Comment (xermicus) [inline fmt/Cargo.toml:3]: Is this still accurate?

File: fmt/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "forge-fmt"
```
Comment (xermicus) [inline fmt/Cargo.toml:2]: I'm thinking, should we name it `solang-fmt` instead? Or do we just copy it over from time to time leaving as much as possible as it is?

File: fmt/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+version = "0.2.0"
+edition = "2021"
+authors = ["Foundry Contributors"]
+license = "MIT OR Apache-2.0"
```
Comment (xermicus) [inline fmt/Cargo.toml:6]: Is this compatible with our license (looks like we have Apache only)?

File: fmt/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+homepage = "https://github.com/foundry-rs/foundry"
+repository = "https://github.com/foundry-rs/foundry"
```
Comment (xermicus) [inline fmt/Cargo.toml:8]: Is this intentional? Might be misleading for example if we want to publish it on crates.io but I guess this isn't supposed to be published?

**Review by xermicus — APPROVED** [review]
LGTM

**Comment by xermicus** [comment]
Yeah I quickly had the same thought as @LucasSte, however we'd have to modify it anyways and it wouldn't be just `forge-fmt` but the whole foundry repo.

**Comment by xermicus** [comment]
@seanyoung also noticed the PR title could be renamed to "Vendor forge-fmt into solang" as it would explain the reasoning of why we need this :)

File: fmt/Cargo.toml (hunk 2 above)
Comment (seanyoung) [reply fmt/Cargo.toml:2]: It's a pretty useful crate TBH. We might want to publish it in the future.

So I agree, why not rename it to `solang-fmt`.

File: fmt/Cargo.toml (hunk 1 above)
Comment (seanyoung) [reply fmt/Cargo.toml:3]: What version should it have if we rename it to `solang-fmt`?

Comment (xermicus) [reply fmt/Cargo.toml:3]: If we are going to publish it, intuitively should be the same version as solang-parser and solang?

Comment (xermicus) [reply fmt/Cargo.toml:3]: However if we just want to vendor it we should keep the original version I think

Comment (seanyoung) [reply fmt/Cargo.toml:3]: How about we just vendor it for now. If we want to publish it, we can always rename it and give it a new version.

**Comment by seanyoung** [comment]
The windows failure is unrelated to this PR

File: fmt/Cargo.toml (hunk 2 above)
Comment (seanyoung) [reply fmt/Cargo.toml:2]: I've changed my mind. Let's just vendor it for now.

File: fmt/Cargo.toml (hunk 4 above)
Comment (seanyoung) [reply fmt/Cargo.toml:8]: As we're vendoring for now, let's leave it as is

**Pull request merged**